File: app/src/main/java/com/example/nijimac103/itunestracker/service/util/FragmentUtils.java
package com.example.nijimac103.itunestracker.service.util;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

import com.example.nijimac103.itunestracker.R;

import static com.google.common.base.Preconditions.checkNotNull;

public class FragmentUtils {

    public static void insertFragmentToActivity(@NonNull FragmentManager fragmentManager,
                                                @NonNull Fragment fragment, int frameId) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(frameId, fragment);
        transaction.commit();
    }

    public static void insertFragmentToActivity(int parentLayoutId, @NonNull FragmentManager fragmentManager,
                                                @NonNull Fragment fragment, String tag) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(parentLayoutId, fragment, tag);
        transaction.commit();
    }

    public static void insertFragmentAddBackStack(int parentLayoutId,
                                                  @NonNull FragmentManager fragmentManager,
                                                  @NonNull Fragment fragment,
                                                  String tag,
                                                  String backStack) {
        checkNotNull(fragmentManager);
        checkNotNull(fragment);
        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction
                .add(parentLayoutId, fragment, tag)
                .addToBackStack(backStack)
                .replace(parentLayoutId, fragment, null)
                .commit();

    }

    public static void setArgsToFragment(Fragment fragment, String key, String value) {
        Bundle bundle = new Bundle();
        bundle.putString(key, value);

        fragment.setArguments(bundle);
    }

    public static String getArgsOfPreFragment(Fragment fragment, String key) {
        Bundle bundle = fragment.getArguments();
        String url = bundle.getString(key);

        return url;
    }
}

File: app/src/main/java/jp/co/geisha/itunestracker/service/util/FragmentUtils.kt
package jp.co.geisha.itunestracker.service.util

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction

import jp.co.geisha.itunestracker.R

import com.google.common.base.Preconditions.checkNotNull

object FragmentUtils {

    fun insertFragmentToActivity(fragmentManager: FragmentManager,
                                 fragment: Fragment, frameId: Int) {
        checkNotNull(fragmentManager)
        checkNotNull(fragment)
        val transaction = fragmentManager.beginTransaction()
        transaction.add(frameId, fragment)
        transaction.commit()
    }

    fun insertFragmentToActivity(parentLayoutId: Int, fragmentManager: FragmentManager,
                                 fragment: Fragment, tag: String) {
        checkNotNull(fragmentManager)
        checkNotNull(fragment)
        val transaction = fragmentManager.beginTransaction()
        transaction.add(parentLayoutId, fragment, tag)
        transaction.commit()
    }

    fun insertFragmentAddBackStack(parentLayoutId: Int,
                                   fragmentManager: FragmentManager,
                                   fragment: Fragment,
                                   tag: String,
                                   backStack: String) {
        checkNotNull(fragmentManager)
        checkNotNull(fragment)
        val transaction = fragmentManager.beginTransaction()
        transaction
                .add(parentLayoutId, fragment, tag)
                .addToBackStack(backStack)
                .replace(parentLayoutId, fragment, null)
                .commit()

    }

    fun setArgsToFragment(fragment: Fragment, key: String, value: String) {
        val bundle = Bundle()
        bundle.putString(key, value)

        fragment.arguments = bundle
    }

    fun getArgsOfPreFragment(fragment: Fragment, key: String): String? {
        val bundle = fragment.arguments!!

        return bundle.getString(key)
    }
}
